pkg/grafana/types.go

```diff
@@ -38,35 +38,61 @@ func NewOrganization(grafanaOrganization *v1alpha1.GrafanaOrganization) Organiza
 	}
 }
 
+// Datasource represents a Grafana datasource.
 type Datasource struct {
-	ID        int64
-	UID       string
-	Name      string
-	IsDefault bool
-	Type      string
-	URL       string
-	Access    string
-	JSONData  map[string]interface{}
-}
-
-func (d Datasource) withID(id int64) Datasource {
-	d.ID = id
-	return d
+	ID             int64
+	UID            string
+	Name           string
+	Type           string
+	URL            string
+	IsDefault      bool
+	Access         string
+	JSONData       map[string]any
+	SecureJSONData map[string]string
 }
 
-func (d Datasource) buildJSONData() map[string]interface{} {
-	if d.JSONData == nil {
-		d.JSONData = make(map[string]interface{})
+// Merge merges the non-zero fields from src into d and returns the result.
+// For JSONData and SecureJSONData maps, it merges the key-value pairs, with src's values taking precedence in case of key conflicts.
+func (d Datasource) Merge(src Datasource) Datasource {
+	if !IsZeroVar(src.ID) {
+		d.ID = src.ID
+	}
+	if !IsZeroVar(src.UID) {
+		d.UID = src.UID
+	}
+	if !IsZeroVar(src.Name) {
+		d.Name = src.Name
+	}
+	if !IsZeroVar(src.Type) {
+		d.Type = src.Type
+	}
+	if !IsZeroVar(src.URL) {
+		d.URL = src.URL
+	}
+	if !IsZeroVar(src.IsDefault) {
+		d.IsDefault = src.IsDefault
+	}
+	if !IsZeroVar(src.Access) {
```

Comment:
So we can never set it to false?

Reply:
Good catch, but I assume you are talking about the `IsDefault` field right ?

Reply:
Yes probably :D

Reply:
Would checkinh against the actual zéro type not bé better with performance though? I assume this fonction will cast types and bé less efficient that s.url != ""?